cmd/ausoceantv/userauth.go

```diff
@@ -0,0 +1,374 @@
+/*
+DESCRIPTION
+
+	Google authentication ("gauth") services using OAuth2.
+
+AUTHORS
+
+	David Sutton <davidsutton@ausocean.org>
+
+LICENSE
+
+	Copyright (C) 2024 the Australian Ocean Lab (AusOcean)
+
+	This is free software: you can redistribute it and/or modify it
+	under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	It is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	in gpl.txt. If not, see http://www.gnu.org/licenses/.
+*/
+package main
+
+import (
+	"context"
+	"encoding/gob"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/ausocean/cloud/gauth"
```

Comment:
We usually order import:
standard lib, then 3rd party, then ausocean packages, with a new line in between.